src/toil/jobStores/googleJobStore.py

```diff
@@ -269,6 +271,8 @@ def readFile(self, jobStoreFileID, localFilePath, symlink=False):
             with open(tmpPath, 'wb') as writeable:
                 blob = self.bucket.get_blob(compat_bytes(jobStoreFileID), encryption_key=self.sseKey)
                 blob.download_to_file(writeable)
+        if getattr(jobStoreFileID, 'executable', False):
+            os.chmod(localFilePath, os.stat(localFilePath).st_mode | stat.S_IXUSR)
```

Comment:
It would be nicer if we could tell AtomicFileCreate whether we wanted an executable file or not. What's the point of atomically creating it with the right contents if the metadata isn't also part of that atomic operation?

Reply:
That's a good point.  I'll look into changing this.

Reply:
I think I'm actually going to merge this and then make an issue to address `AtomicFileCreate` separately.

Reply:
#3467